### DISCUSSION TITLE
Network drive acces

### DISCUSSION THREAD

**Discussion by JuanDeDironne** [coreybutler/node-windows#326] [General]
Hello
First of all thank you for this mod, it's exactly what I needed.

I developed a Node JS application that allows you to browse a folder and then send a file from this folder in FTP. If I run my application with the following command: node application.js :
- I can browse the contents of a folder locally
- I can browse the contents of a folder of a network drive

If I run my application through a service :
- I can browse the contents of a folder locally
- I can't browse the contents of a folder of a network drive
I get an error `W:\my_folder\ Error: ENOENT: no such file or directory, scandir 'W:\my_folder\'`

Does anyone have any idea why my application running as a service can't browse network drives...?

**Comment by miquelfire** [comment]
Your W: drive is a mapped drive, so it will only appear for the logged user while they're on the desktop (and in non-admin sessions if you left UAC enabled, it doesn't appear for the elevated sessions when something needs admin rights), you'll need to use the direct path, and I'm not sure how you'll do it in node.js, but also may need to provide the credentials for the path as well.

**Reply by JuanDeDironne** [reply]
It seems fine to me to use absolute paths but I'm doing some new tests and coming back to you just in case.

**Comment by coreybutler** [comment]
As @miquelfire stated, the user running the service needs read access to the `W:\my_folder\....` path. 

If the account running the service does have permissions, there are other some other general things to be aware of. The [wiki "My script runs fin on it's own but not with node-windows" section](https://github.com/coreybutler/node-windows/wiki#my-script-runs-fine-on-its-own-but-not-with-node-windows-why) may provide help.

**Reply by JuanDeDironne** [reply]
Thank you for your answers. I read, I test and I come back to you with new information

**Reply by JuanDeDironne** [reply]
I read your link and I coded a very small application with absolute paths. I use the "`fs.readdirSync()`" function for these tests.
And I have the same problem. In the command line everything works but not since with the launch as a Service.
In the list of services, I find in the column "Log in as" the following information "Local System".
The problem would therefore be a problem of law. The service cannot access the network drive (while for local drives including USB drives everything works)

**Reply by JuanDeDironne** [reply]
And if I want to specify a user for this service I can use this code...?
svc.logOnAs.domain = 'mydomain.local';
svc.logOnAs.account = 'username';
svc.logOnAs.password = 'password';

**Comment by JuanDeDironne** [comment]
I continued my tests in order to better understand.
I created a simple application for testing and in this application I plotted user information (with `os.userInfo()` and module os).
If I run in console mode I get this info :
name : factice
homedir : C:\Users\factice

If I run in service mode (just after creating the service) I have this information :
name : System
homedir : C:\WINDOWS\system32\config\systemprofile

As I have a difference I modified in the services management console the session connection parameter to use the same user as in command mode.
Then if I restart the service I have this information :
name : factice
homedir : C:\Users\factice
But even if the information is identical. Network drive browse does not work.

**Reply by miquelfire** [reply]
Anything that appears in the network locations in the "This PC" view in file explorer will not be accessible from a service. Your app will have to use the URL you're mapped to. So if W: is mapped to \\server\myshare, then your app will need to use \\server\myshare and not W:

Note: In my example, I'm assuming that W: is mapped to a network share, but your first post talked about FTP, and I haven't messed with that on Windows, so I don't know how the URL for that would look like or how you'll use it in Node.js

**Reply by JuanDeDironne** [reply]
As for FTP. I just use the FTP protocol to deposit a file present on a local drive or network drive.
Just for that. My real problem is to browse the network drive (where the file I will send later via FTP is located)

And we agree, I had the same thought as you on the shortcut with the letter and the full path. And so I tested with the full path. But it still doesn't work :(

And I discovered that if we use a simple terminal we have access to network drives but if we launch a terminal in administrator mode we do not have access to network drives (with exactly the same errors obtained via the application in Service mode ).

Issue due to drives not being shared between sessions. And this problem I have just solved by adding the key "**EnableLinkedConnections**" in the registry.

But I haven't tested the impact on the Node JS application in Service mode yet. I will continue my research and come back to you. And thank you for your help.
